File: Assignment38_5.cpp
// 5. Write a program which accept one number from user and on its first 4 
// bits. Return modified number.

#include<iostream>
using namespace std;

class BitOpration
{
    public:
        int iNo;
        int iMask;
        int iResult;

        BitOpration(int A)
        {
            iNo = A;
            iMask = 0X0000000f;
            iResult = 0;
        }

        int OffBit()
        {
            iResult = iNo & iMask;

            if(iResult == iMask) //Bit is On
            {
                return iNo;
            }
            else//Bit is off
            {
                
                return (iNo | iMask);
            }
        }


};

int main()
{
    int iValue = 0;
    int iRet = 0;
    
    cout<<"Enter the Number Here : \n";
    cin>>iValue;

    BitOpration bobj(iValue);
    iRet = bobj.OffBit();

        
    cout<<"Modified Number is : "<<iRet<<"\n";
    

    return 0;
}

File: Assignment38_5.py
# 5. Write a program which accept one number from user and on its first 4 
# bits. Return modified number.


class BitOpeartion:
    iValue = None
    iMask = None
    iResult = None
    
    def __init__(self,iValue):
        self.iValue = iValue
        self.iMask = 0X0000000f
        self.iResult = 0

    def ToggleBit(self):
        
        self.iResult = self.iValue & self.iMask
        
        if(self.iResult == self.iMask):
            return (self.iValue ^ self.iMask)
        else:
            return (self.iValue ^ self.iMask)

def main():
    iNo = 0
    iRet = 0
    
    print("Enter the Number Here : ")
    iNo = int(input())
    
    Obj = BitOpeartion(iNo)
    
    iRet = Obj.ToggleBit()
    
    print("The Modified Number after Toggling Bit is : ",iRet)
    
if __name__=="__main__":
    main()